coursier.bzl

```diff
@@ -31,12 +34,16 @@ load("@{repository_name}//:jvm_import.bzl", "jvm_import")
 # download them if it's not asked to to resolve "eclipse-plugin".
 _COURSIER_PACKAGING_TYPES = ["jar", "aar", "bundle", "eclipse-plugin"]
 
-# Super hacky :(
-def _strip_packaging_and_classifier(coord):
-    return coord.replace(":jar:", ":").replace(":aar:", ":").replace(":sources:", ":")
+def _strip_packaging_classifier(coord):
+    for packaging_type in _COURSIER_PACKAGING_TYPES + ["pom"]:
+        coord = coord.replace(":%s:" % packaging_type, ":")
+    for classifier_type in ["sources", "natives"]:
+        coord = coord.replace(":%s:" % classifier_type, ":")
+
+    return coord
 
-def _strip_packaging_and_classifier_and_version(coord):
-    return ":".join(_strip_packaging_and_classifier(coord).split(":")[:-1])
+def _strip_packaging_classifier_version(coord):
+    return ":".join(_strip_packaging_classifier(coord).split(":")[:-1])
```

Comment:
Why the change to the name? _strip_packaging_and_classifier is clearer (without the and it sounds like you're stripping a single thing called "packaging classifier"). Also with the version method

Reply:
Reverted.